src/shared/extension-release.c

```diff
@@ -18,7 +18,11 @@ int extension_release_validate(
         const char *extension_release_id = NULL, *extension_release_sysext_level = NULL;
 
         assert(name);
-        assert(!isempty(host_os_release_id));
+
+        if (isempty(host_os_release_id)) {
+                log_debug("Extension '%s' does not have valid ID, ignoring extension.", name);
+                return 0;
```

Comment:
this should be a hard error, not an ignore, because ID is mandatory - it needs to be propagated back to the caller

Reply:
also this is not the extension's, this is the host's?

Reply:
I'll send a PR shortly with an alternative fix

Reply:
https://github.com/systemd/systemd/pull/23452